### PR TITLE
Included version information which the CLI starts up

### PR DESCRIPTION
**Pull request by naveensrinivasan**
# Description of the PR

- Included the version information when the CLI starts up.

```
{"level":"info","ts":1697396471.853624,"caller":"version/version.go:57","msg":"Commit: dev-7c3b1b9188e868c1ce2c8c21793cd7e6894aa244\n"}
{"level":"info","ts":1697396471.853778,"caller":"version/version.go:64","msg":"Date: 2023-10-15"}
```

# PR Checklist

- [x] All commits have [a Developer Certificate of Origin (DCO)](https://wiki.linuxfoundation.org/dco) -- they are generated using `-s` flag to `git commit`.
- [ ] All new changes are covered by tests
- [ ] If GraphQL schema is changed, `make generate` has been run
- [ ] If `collectsub` protobuf has been changed, `make proto` has been run
- [ ] All CI checks are passing (tests and formatting)
- [ ] All dependent PRs have already been merged

### CODE REVIEW

File: cmd/guaccollect/cmd/deps_dev.go [hunk 1]
```diff
@@ -53,6 +54,8 @@ var depsDevCmd = &cobra.Command{
 		ctx := logging.WithLogger(context.Background())
 		logger := logging.FromContext(ctx)
 
+		version.DumpVersion()
```
Comment (lumjjb) [inline cmd/guaccollect/cmd/deps_dev.go:57]: i dont think we need to dump the version on every invocation. I think CLIs can have a `--version` flag which should output the version,

File: pkg/version/version.go [hunk 2]
```diff
@@ -40,3 +46,23 @@ func (u uat) RoundTrip(r *http.Request) (*http.Response, error) {
 	r.Header.Set("User-Agent", UserAgent)
 	return u.tr.RoundTrip(r)
 }
+
+// DumpVersion dumps the version information to the log.
+func DumpVersion() {
```
Comment (lumjjb) [inline pkg/version/version.go:51]: this implementation needs to be robust , as it stands it is not. Plus it depends on a CLI command.. which I don't want us to have a dependency for this trade-off. I think a better more robust solution is needed here. I believe golang has some in built features to do this.

**Review by lumjjb — CHANGES_REQUESTED** [review]

File: cmd/guaccollect/cmd/deps_dev.go (hunk 1 above)
Comment (naveensrinivasan) [reply cmd/guaccollect/cmd/deps_dev.go:57]: The goal was to know which version of the CLI runs when the server starts. 

Right now, that isn't possible.  

That was the reason for this change.

Do you have suggestions for that?

File: pkg/version/version.go (hunk 2 above)
Comment (naveensrinivasan) [reply pkg/version/version.go:51]: It is using already using builtin https://github.com/guacsec/guac/blob/7c3b1b9188e868c1ce2c8c21793cd7e6894aa244/.goreleaser.yaml#L67-L69

If the built-in fails, we use the commit SHA and the present date and run the non-released version.

Let me know if you have a better strategy to solve this issue.

**Pull request closed without merging**